pep-0688.rst

```diff
@@ -203,30 +207,43 @@ buffer protocol:
 
 .. code-block:: python
 
+   import contextlib
+   import inspect
+
    class MyBuffer:
        def __init__(self, data: bytes):
            self.data = bytearray(data)
-           self.view = memoryview(self.data)
-           self.held = False
+           self.view = None
 
        def __buffer__(self, flags: int) -> memoryview:
            if flags != inspect.BufferFlags.FULL_RO:
                raise TypeError("Only BufferFlags.FULL_RO supported")
-           if self.held:
+           if self.view is not None:
                raise RuntimeError("Buffer already held")
-           self.held = True
+           self.view = memoryview(self.data)
            return self.view
 
        def __release_buffer__(self, view: memoryview) -> None:
-           assert self.view is view
-           self.held = False
+           assert self.view is view  # guaranteed to be true
```

Comment:
As in previous comment. I think this is *not* guaranteed when a C client uses the C-API (and therefore the slot method wrapping this).

Reply:
Resolved for me as in previous comment.